src/accessibility/outputs.js

```diff
@@ -0,0 +1,418 @@
+ *   //fill(0, 0, 255);
+ *   //rect(50, 50, 50, 50);
+ *   //ellipse(20, 20, 20, 20);
+ *   //x += 0.1;
+ * //}
+ *
+ */
+
+p5.prototype.gridOutput = function(display) {
+  p5._validateParameters('gridOutput', arguments);
+  if (txtOut === false) {
+    grOut = true;
+    if (this.canvas !== undefined) {
+      this._setDefaults();
+      this._createOutput('grOut');
+    } else {
+      throw new Error('gridOutput() should be called after canvas is created');
+    }
+  } else {
+    return;
+  }
+};
+
+//helper function returns true when accessible outputs are true
+p5.prototype._addAccsOutput = function() {
+  if (txtOut === true || grOut === true) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+//helper function that creates html structure for accessible outputs
+p5.prototype._createOutput = function(type) {
+  let cnvId = this.canvas.id;
+  let cIdT = cnvId + type;
+  if (document.getElementById(cIdT) === null) {
+    document
+      .getElementById(cnvId)
+      .insertAdjacentHTML(
+        'afterend',
+        '<section id="' + cIdT + '" class="accessibleOutput"></section>'
+      );
+    if (type === 'txtOut') {
+      let inner = this._createTextOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
+    }
+    if (type === 'grOut') {
+      let inner = this._createGridOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
+    }
+    //this._updateOutput();
+  }
+};
+
+//helper function that updates accessible outputs
+p5.prototype._updateOutput = function() {
+  if (JSON.stringify(ingredients) !== JSON.stringify(preIngredients)) {
+    this._updateOutput();
+  }
+  let cnvId = this.canvas.id;
+  if (txtOut === true) {
+    this._updateTextOutput(cnvId, ingredients, cnvConfig.background);
+  }
+  if (grOut === true) {
+    this._updateGridOutput(cnvId, ingredients, cnvConfig.background);
+  }
+};
+
+//helper function that resets all ingredients when background is called
+//and saves background color name
+p5.prototype._accsBackground = function(args) {
+  preIngredients = ingredients;
+  ingredients = {};
+  if (cnvConfig.backgroundRGBA !== args) {
+    //cnvConfig.backgroundRGBA = args;
+    cnvConfig.background = this._rgbColorName(args);
+  }
+};
+
+//helper function that gets fill and stroke of shapes
+p5.prototype._accscnvConfig = function(f, args) {
+  if (f === 'fill') {
+    if (cnvConfig.fillRGBA !== args) {
+      cnvConfig.fillRGBA = args;
+      cnvConfig.fill = this._rgbColorName(args);
+    }
+  } else if (f === 'stroke') {
+    if (cnvConfig.strokeRGBA !== args) {
+      cnvConfig.strokeRGBA = args;
+      cnvConfig.stroke = this._rgbColorName(args);
+    }
+  }
+};
+
+//helper function that sets defaul colors for background
+//fill and stroke.
+p5.prototype._setDefaults = function() {
+  cnvConfig.background = 'white';
+  cnvConfig.fill = 'white';
+  cnvConfig.stroke = 'black';
+};
+
+// return length of lines
+p5.prototype._getLineL = function(args) {
```

Comment:
consider `_getLineLength` for readability.
but, does the existing `[dist](https://p5js.org/reference/#/p5/dist)` function work for this?